src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -344,13 +343,14 @@ vector<SparqlTriple> Visitor::visitTypesafe(Parser::TriplesBlockContext* ctx) {
   auto triples = ad_utility::transform(
       visitTypesafe(ctx->triplesSameSubjectPath()), convertAndRegisterTriple);
   if (ctx->triplesBlock()) {
-    ad_utility::appendVector(triples, visitTypesafe(ctx->triplesBlock()));
+    ad_utility::appendVector(
+        triples, visitTypesafe(ctx->triplesBlock())._whereClauseTriples);
```

Comment:
If you change the type of `triples` to be `BasicGraphPattern`, then you can reuse the `appendTriples` method I have suggested above.